osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -53,54 +60,79 @@ public ISkin FindProvider(Func<ISkin, bool> lookupFunction)
             }
             else if (skin != null)
             {
-                if (lookupFunction(skin))
+                // a proxy must be used here to correctly pass through the "Allow" checks without implicitly falling back to the fallbackSource.
+                if (lookupFunction(noFallbackLookupProxy))
```

Comment:
I'm so confused. How does this in any way "pass through the Allow checks"?

```
noFallbackLookupProxy = new NoFallbackProxy(this);

noFallbackLookupProxy.GetTexture(...)
  == provider .GetTexture(..., false)
  == this.GetTexture(..., false)
```
What am I missing?

Reply:
by "pass through" i mean "run" them.

the critical part of that comment is the following words "without implicitly falling back to the fallbackSource"

Reply:
I'm not sure why, but having public methods with the fallback parameter confused me.

Reply:
Do you think it's worth tidying them up? Moving the interface implementation to implicit and giving `fallback` a default parameter?

Reply:
Dunno, it's fine for now. Probably makes more sense this way from the consumer's POV.